### PR TITLE
Allow the PUT method globally in the deck app.

### PR DESCRIPTION
**Pull request by mivek**
I updated the rule added in #53 to allow to `PUT` method when moving cards, un-assigning users and setting a due date.

### CODE REVIEW

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 1]
```diff
@@ -1531,8 +1531,8 @@ SecRule REQUEST_FILENAME "@rx /ocs/v[0-9]+\.php/apps/user_status/api/v[0-9]+/use
 # [ Nextcloud Deck ]
 #
 
-# Moving a card in Deck app
-SecRule REQUEST_FILENAME "@rx /apps/deck/cards/[0-9]+/reorder$" \
+# When updating a card
+SecRule REQUEST_FILENAME "@rx /apps/deck/cards/[0-9]+" \
```
Comment (EsadCetiner) [inline plugins/nextcloud-rule-exclusions-before.conf:1535]: You should try to match as exact as you can for the URL path to minimize the chance of a bypass. Based on the URLs provided in your regression test, you should use this regular expression.
```suggestion
# When updating a card
# Moving a card in deck
# Unassigning a user from a card
# Setting a due date on a card
SecRule REQUEST_FILENAME "@rx /apps/deck/cards/[0-9]+(?:/reorder|/unassign)?$" \
```
https://regexr.com/7s4pl

**Review by EsadCetiner — CHANGES_REQUESTED** [review]

**Review by EsadCetiner — APPROVED** [review]

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 2]
```diff
@@ -27,3 +27,41 @@ tests:
...
+              Host: localhost
+              User-Agent: OWASP ModSecurity Core Rule Set
+              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+            port: 80
+            method: PUT
+            uri: /apps/deck/cards/1/unassign
+          output:
+            no_log_contains: |
+              id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"
+  - test_title: 9508810-2
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:48]:
```suggestion
  - test_title: 9508810-3
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 3]
```diff
@@ -27,3 +27,41 @@ tests:
...
+    desc: |
+      Setting a due date to a card.
+      Target: description
+    stages:
+      - stage:
+          input:
+            dest_addr: 127.0.0.1
+            headers:
+              Host: localhost
+              User-Agent: OWASP ModSecurity Core Rule Set
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:58]:
```suggestion
              User-Agent: OWASP CRS
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 4]
```diff
@@ -27,3 +27,41 @@ tests:
...
+    desc: |
+      Unassigning a user from a card.
+      Target: description
+    stages:
+      - stage:
+          input:
+            dest_addr: 127.0.0.1
+            headers:
+              Host: localhost
+              User-Agent: OWASP ModSecurity Core Rule Set
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:40]:
```suggestion
              User-Agent: OWASP CRS
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 5]
```diff
@@ -27,3 +27,41 @@ tests:
           output:
             no_log_contains: |
               id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"
+  - test_title: 9508810-2
+    desc: |
+      Unassigning a user from a card.
+      Target: description
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:33]:
```suggestion
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 6]
```diff
@@ -27,3 +27,41 @@ tests:
           output:
             no_log_contains: |
               id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"
+  - test_title: 9508810-2
+    desc: |
+      Unassigning a user from a card.
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:32]:
```suggestion
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 7]
```diff
@@ -27,3 +27,41 @@ tests:
           output:
             no_log_contains: |
               id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"
+  - test_title: 9508810-2
+    desc: |
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:31]:
```suggestion
    desc: Unassigning a user from a card
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 8]
```diff
@@ -27,3 +27,41 @@ tests:
...
+            port: 80
+            method: PUT
+            uri: /apps/deck/cards/1/unassign
+          output:
+            no_log_contains: |
+              id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"
+  - test_title: 9508810-2
+    desc: |
+      Setting a due date to a card.
+      Target: description
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:51]:
```suggestion
      Target: json.data
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml [hunk 9]
```diff
@@ -27,3 +27,41 @@ tests:
...
+              Accept: text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5
+            port: 80
+            method: PUT
+            uri: /apps/deck/cards/1/unassign
+          output:
+            no_log_contains: |
+              id "911100"|id "932200"|id "933210"|id "942200"|id "942260"|id "942370"|id "942430"|id "942440"
+  - test_title: 9508810-2
+    desc: |
+      Setting a due date to a card.
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508810.yaml:50]:
```suggestion
      Setting a due date on a card.
```

**Review by theseion — APPROVED** [review]

**Pull request merged**